tests/api/patrons/test_patrons_rest.py

```diff
@@ -552,3 +552,18 @@ def test_patrons_count(client, patron_sion_no_email,
     res = client.get(url)
     assert res.status_code == 200
     assert get_json(res) == dict(hits=dict(total=1))
+
+
+def test_patrons_circulation_stats(client, patron_sion_no_email,
+                                   librarian_martigny_no_email):
+    """test patron circulation stats."""
+    url = url_for(
+        'api_patrons.patron_circulation_stats',
+        patron_pid=patron_sion_no_email.pid
+    )
+    res = client.get(url)
+    assert res.status_code == 401
+
+    login_user_via_session(client, librarian_martigny_no_email.user)
+    res = client.get(url)
+    assert res.status_code == 200
```

Comment:
Can you add a test to validate the output of the function `get_loans_stats_by_patron_pid`? so we are sure it is returning the expected results

Reply:
It's difficult because stats are different if you execute the test as standalone or with the full tests.